SimG4CMS/Calo/src/HFShowerLibrary.cc

```diff
@@ -211,6 +211,21 @@ std::vector<HFShowerLibrary::Hit> HFShowerLibrary::getHits(G4Step * aStep,
   G4String      partType = track->GetDefinition()->GetParticleName();
   int           parCode  = track->GetDefinition()->GetPDGEncoding();
 
+#ifdef DebugLog
+  G4ThreeVector localPos = preStepPoint->GetTouchable()->GetHistory()->GetTopTransform().TransformPoint(hitPoint);
+  double zoff   = localPos.z() + 0.5*gpar[1];
+  //  if (zoff < 0) zoff = 0;
+  edm::LogInfo("HFShower") << "HFShowerLibrary: getHits " << partType
+                           << " of energy " << pin/GeV << " GeV"
+                           << "  dir.orts " << momDir.x() << ", " <<momDir.y()
+                           << ", " << momDir.z() << "  Pos x,y,z = "
+                           << hitPoint.x() << "," << hitPoint.y() << ","
+                           << hitPoint.z() << " (" << zoff
+                           << ")   sphi,cphi,stheta,ctheta  = " << sin(momDir.phi())
+                           << ","  << cos(momDir.phi()) << ", " << sin(momDir.theta()) 
+                           << "," << cos(momDir.theta());
+#endif
+
   std::vector<HFShowerLibrary::Hit> hit;
   ok = false;
   if (parCode == pi0PDG || parCode == etaPDG || parCode == nuePDG ||
```

Comment:
@civanch , is there anything against initialising the *PDG variables in the constructor to the usual PDG codes?
@Vlandr57 , could you take care of that?